### PR TITLE
ci: drop cpp20 docker image

### PR DESCRIPTION
**Pull request by vvbandeira**

### CODE REVIEW

**Comment by vvbandeira** [comment]
@maliberty @gadfort FYI

File: etc/DependencyInstaller.sh [hunk 1]
```diff
@@ -696,6 +698,12 @@ _installCI() {
         docker-ce-cli \
         containerd.io \
         docker-buildx-plugin
+
+    # Install clang for C++20 support
+    wget https://apt.llvm.org/llvm.sh
+    chmod +x llvm.sh
+    ./llvm.sh 16 all
```
Comment (maliberty) [inline etc/DependencyInstaller.sh:705]: Will this cause a change in compiler version for non-c++20 builds?

**Review by maliberty — APPROVED** [review]

File: etc/DependencyInstaller.sh (hunk 1 above)
Comment (vvbandeira) [reply etc/DependencyInstaller.sh:705]: No, by default we compile with gcc you need to ask cmake or the `Build.sh` script to use `clang-16` or other compilers.

Comment (maliberty) [reply etc/DependencyInstaller.sh:705]: ok - ready to merge?

Comment (vvbandeira) [reply etc/DependencyInstaller.sh:705]: On my side yes, not sure if Peter depended on this Dockerfile for anything or if it was just for your internal testing.

**Pull request merged**